File: 49.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        dic = dict()
        for item in strs:
            temp = ''.join(sorted(item))
            dic[temp] = dic.get(temp, []) + [item]
        return dic.values()

File: 49.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, int> m;
        vector<vector<string>> res;
        for(auto str: strs){
            string s = str;
            std::sort(s.begin(), s.end());
            if (!m.count(s)){
                m[s] = res.size();
                res.push_back({});
            }
            res[m[s]].push_back(str);
        }
        return res;
    }
};
